internal/envoy/cluster.go

```diff
@@ -183,6 +215,18 @@ func Clustername(cluster *dag.Cluster) string {
 	return hashname(60, ns, name, strconv.Itoa(int(service.Weighted.ServicePort.Port)), fmt.Sprintf("%x", hash[:5]))
 }
 
+// ExtensionClusterName generates a unique Envoy cluster name for an
+// ExtensionService. ExtensionService resources are globally unique,
```

Comment:
nit: They aren't globally unique since they are scoped to a namespace. A cluster CRD would be globally unique. I know what you are getting at but they are unique with respect to namespace+name.

Reply:
Yeh I meant that the namespace+name is globally unique. I can probably phrase this better :)